File: answers/leetcode/Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
	int removeDuplicates(int A[], int n) {
		if (0 == n) {
			return n;
		}
		int duplicate = A[0];
		int count = 0;
		for (int i = 0; i < n; ++ i) {
			if (A[i] == duplicate) {
				++ count;
				if (count > 2) {
					for (int j = i + 1; j < n; ++ j) {
						A[j - 1] = A[j];
					}
					-- i;
					-- n;
				}
			} else {
				count = 1;
				duplicate = A[i];
			}
		}
		return n;
	}
};

File: answers/leetcode/Remove Duplicates from Sorted Array II.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        prev0 = 0
        prev1 = 1
        for i in range(2, len(nums)):
            if nums[i] != nums[prev0] or nums[i] != nums[prev1]:
                prev0 += 1
                prev1 += 1
                nums[prev1] = nums[i]
        return min(len(nums), prev1 + 1)
